Support Linux

what files exactly are missing?
I just tried to build an example and encountered:
```txt
/usr/bin/ld: cannot find -lfmodstudio: No such file or directory
/usr/bin/ld: cannot find -lfmod: No such file or directory
collect2: error: ld returned 1 exit status
```
which I presume is the same issue you found. I'm not sure where these should be coming from.

Is this an issue inherited from libfmod? Which files are missing?

--- a/.gitignore
+++ b/.gitignore
@@ -2,5 +2,7 @@ target/
 assets/
 *.lib
 *.dll
+*.so
+*.so.*
 demo_project/
 Cargo.lock

--- a/README.md
+++ b/README.md
@@ -43,7 +43,9 @@ This requires a free FMOD account.
     - `api/studio/lib/x64/fmodstudio_vc.lib`: **rename to `fmodstudio.lib`**
 
 ### Linux
+
 Below are the steps for a fairly minimal method to link the libraries. See the comments in [build.rs](https://github.com/Salzian/bevy_fmod/blob/main/build.rs) for more information.
+
 - Download the "FMOD Engine" package for Linux. Make sure to select [a compatible version](#versioning).
 - Create a new folder `fmod` in the root of your project.
 - Extract the `api` folder into it.

--- a/README.md
+++ b/README.md
@@ -43,11 +43,11 @@ This requires a free FMOD account.
     - `api/studio/lib/x64/fmodstudio_vc.lib`: **rename to `fmodstudio.lib`**
 
 ### Linux
+Below are the steps for a fairly minimal method to link the libraries. See the comments in [build.rs](https://github.com/Salzian/bevy_fmod/blob/main/build.rs) for more information.
 - Download the "FMOD Engine" package for Linux. Make sure to select [a compatible version](#versioning).
 - Create a new folder `fmod` in the root of your project.
 - Extract the `api` folder into it.
 - Copy the contents of [build.rs](https://github.com/Salzian/bevy_fmod/blob/main/build.rs) into your own build script.
-- Note that this is just an example, you are free to setup the folders and build methods as you wish.
 
 ## Usage
 

--- a/README.md
+++ b/README.md
@@ -47,6 +47,7 @@ This requires a free FMOD account.
 - Create a new folder `fmod` in the root of your project.
 - Extract the `api` folder into it.
 - Copy the contents of [build.rs](https://github.com/Salzian/bevy_fmod/blob/main/build.rs) into your own build script.
+- Note that this is just an example, you are free to setup the folders and build methods as you wish.
 
 ## Usage
 

--- a/README.md
+++ b/README.md
@@ -20,10 +20,10 @@ Learn more [here][FMOD attribution].
 
 ## Supported platforms
 
-Currently, this crate is only tested and developed for Windows (non-UWP). More platforms are planned eventually.
+Currently, this crate is only tested and developed for Windows (non-UWP) and Linux. More platforms are planned eventually.
 
 MacOS: https://github.com/Salzian/bevy_fmod/issues/2  
-Linux: https://github.com/Salzian/bevy_fmod/issues/3
+Web: https://github.com/Salzian/bevy_fmod/issues/51
 
 Pull requests are welcome.
 
@@ -33,6 +33,7 @@ This crate does not bundle the required FMOD libraries. You will need to downloa
 libraries [here][FMOD libraries download].
 This requires a free FMOD account.
 
+### Windows
 - Download the "FMOD Engine" package for Windows. Make sure to select [a compatible version](#versioning).
 - Install the package.
 - You need the following 4 files in the root of your rust project:
@@ -40,6 +41,15 @@ This requires a free FMOD account.
     - `api/core/lib/x64/fmod_vc.lib`: **rename to `fmod.lib`**
     - `api/studio/lib/x64/fmodstudio.dll`
     - `api/studio/lib/x64/fmodstudio_vc.lib`: **rename to `fmodstudio.lib`**
+
+### Linux
+- Download the "FMOD Engine" package for Linux. Make sure to select [a compatible version](#versioning).
+- Create a new folder `fmod-api` in `/usr/lib/<target_arch>/`.
+- Extract the contents from the following FMOD Engine folders into it:
+  - `api/core/lib/<target_arch>/`
+  - `api/studio/lib/<target_arch>/`
+- Copy the contents of [build.rs](https://github.com/Salzian/bevy_fmod/blob/main/build.rs) into your own build script.
+- Note that currently this build script will not work together with Bevy's dynamic_linking feature.
 
 ## Usage
 

--- a/README.md
+++ b/README.md
@@ -44,12 +44,9 @@ This requires a free FMOD account.
 
 ### Linux
 - Download the "FMOD Engine" package for Linux. Make sure to select [a compatible version](#versioning).
-- Create a new folder `fmod-api` in `/usr/lib/<target_arch>/`.
-- Extract the contents from the following FMOD Engine folders into it:
-  - `api/core/lib/<target_arch>/`
-  - `api/studio/lib/<target_arch>/`
+- Create a new folder `fmod` in the root of your project.
+- Extract the `api` folder into it.
 - Copy the contents of [build.rs](https://github.com/Salzian/bevy_fmod/blob/main/build.rs) into your own build script.
-- Note that currently this build script will not work together with Bevy's dynamic_linking feature.
 
 ## Usage
 

--- a/build.rs
+++ b/build.rs
@@ -1,29 +1,25 @@
 /// This build script is needed to run the examples on Linux.
 /// Feel free to copy this to your own Bevy project.
-/// Note that currently it is not possible to use it together with the Bevy dynamic_linking feature.
+///
+/// Be aware that on Linux you also need to set LD_LIBRARY_PATH to include the libraries.
+/// There are a few ways to do it:
+///
+/// If you are working with an IDE it's easiest to add this to the run configuration.
+/// For example, in Rust Rover you set `Environment variables` to `LD_LIBRARY_PATH=./fmod/api/core/lib/x86_64:./fmod/api/studio/lib/x86_64`.
+///
+/// If you are running your executable directly (no IDE, no cargo) see https://www.hpc.dtu.dk/?page_id=1180
 
 fn main() {
-    #[cfg(all(target_os = "linux", target_arch = "x86"))]
+    #[cfg(target_os = "linux")]
     {
-        println!("cargo:rustc-link-search=/usr/lib/i386-linux-gnu/fmod-api/");
-        println!("cargo:rustc-env=LD_LIBRARY_PATH=/usr/lib/i386-linux-gnu/fmod-api/");
-    }
-
-    #[cfg(all(target_os = "linux", target_arch = "x86_64"))]
-    {
-        println!("cargo:rustc-link-search=/usr/lib/x86_64-linux-gnu/fmod-api/");
-        println!("cargo:rustc-env=LD_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu/fmod-api/");
-    }
-
-    #[cfg(all(target_os = "linux", target_arch = "arm"))]
-    {
-        println!("cargo:rustc-link-search=/usr/lib/arm-linux-gnueabihf/fmod-api/");
-        println!("cargo:rustc-env=LD_LIBRARY_PATH=/usr/lib/arm-linux-gnueabihf/fmod-api/");
-    }
-
-    #[cfg(all(target_os = "linux", target_arch = "aarch64"))]
-    {
-        println!("cargo:rustc-link-search=/usr/lib/aarch64-linux-gnu/fmod-api/");
-        println!("cargo:rustc-env=LD_LIBRARY_PATH=/usr/lib/aarch64-linux-gnu/fmod-api/");
+        let target_arch = std::env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+        let api_loc = [
+            format!("./fmod/api/core/lib/{target_arch}"),
+            format!("./fmod/api/studio/lib/{target_arch}"),
+        ];
+        for loc in api_loc {
+            println!("cargo:rustc-link-search={loc}");
+            println!("cargo:rustc-env=LD_RUN_PATH={loc}");
+        }
     }
 }

--- a/build.rs
+++ b/build.rs
@@ -1,0 +1,29 @@
+/// This build script is needed to run the examples on Linux.
+/// Feel free to copy this to your own Bevy project.
+/// Note that currently it is not possible to use it together with the Bevy dynamic_linking feature.
+
+fn main() {
+    #[cfg(all(target_os = "linux", target_arch = "x86"))]
+    {
+        println!("cargo:rustc-link-search=/usr/lib/i386-linux-gnu/fmod-api/");
+        println!("cargo:rustc-env=LD_LIBRARY_PATH=/usr/lib/i386-linux-gnu/fmod-api/");
+    }
+
+    #[cfg(all(target_os = "linux", target_arch = "x86_64"))]
+    {
+        println!("cargo:rustc-link-search=/usr/lib/x86_64-linux-gnu/fmod-api/");
+        println!("cargo:rustc-env=LD_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu/fmod-api/");
+    }
+
+    #[cfg(all(target_os = "linux", target_arch = "arm"))]
+    {
+        println!("cargo:rustc-link-search=/usr/lib/arm-linux-gnueabihf/fmod-api/");
+        println!("cargo:rustc-env=LD_LIBRARY_PATH=/usr/lib/arm-linux-gnueabihf/fmod-api/");
+    }
+
+    #[cfg(all(target_os = "linux", target_arch = "aarch64"))]
+    {
+        println!("cargo:rustc-link-search=/usr/lib/aarch64-linux-gnu/fmod-api/");
+        println!("cargo:rustc-env=LD_LIBRARY_PATH=/usr/lib/aarch64-linux-gnu/fmod-api/");
+    }
+}

--- a/build.rs
+++ b/build.rs
@@ -12,7 +12,13 @@
 fn main() {
     #[cfg(target_os = "linux")]
     {
-        let target_arch = std::env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+        let mut target_arch = std::env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+
+        // Account for the naming mismatch between FMOD library folders and the target architecture
+        if target_arch == "aarch64" {
+            target_arch = "arm64".into();
+        }
+
         let api_loc = [
             format!("./fmod/api/core/lib/{target_arch}"),
             format!("./fmod/api/studio/lib/{target_arch}"),

